activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -579,13 +579,14 @@ def translate_exception_class(e, sql)
         exception
       end
 
-      def log(sql, name = "SQL", binds = [], statement_name = nil)
+      def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name = nil)
         @instrumenter.instrument(
           "sql.active_record",
           :sql            => sql,
           :name           => name,
           :connection_id  => object_id,
           :statement_name => statement_name,
+          :type_casted_binds => type_casted_binds,
```

Comment:
Could we realign here?

Also, it would feel natural to set the keys in the same order as the corresponding arguments in the method call (including moving `:statement_name` as last key).